base/applications/rapps/gui.cpp

```diff
@@ -2272,17 +2583,36 @@ class CMainWindow :
         case ID_INSTALL:
             if (IsAvailableEnum(SelectedEnumType))
             {
-                if (nSelectedApps > 0)
+                ATL::CSimpleArray<CAvailableApplicationInfo> AppsList;
+
+                // enum all selected apps
+                m_AvailableApps.Enum(ENUM_CAT_SELECTED, s_EnumSelectedAppForDownloadProc, (PVOID)&AppsList);
+
+
+                if (AppsList.GetSize())
                 {
-                    DownloadListOfApplications(m_AvailableApps.GetSelected(), FALSE);
-                    UpdateApplicationsList(-1);
-                    m_ListView->SetSelected(-1, FALSE);
+                    if (DownloadListOfApplications(AppsList, FALSE))
+                    {
+                        m_AvailableApps.RemoveAllSelected();
+                        UpdateApplicationsList(-1);
+                    }
                 }
-                else if (DownloadApplication(m_ListView->GetSelectedData(), FALSE))
+                else
                 {
-                    UpdateApplicationsList(-1);
+                    // use the currently focused item in tableview
+                    CAvailableApplicationInfo *FocusedApps = (CAvailableApplicationInfo *)m_AppsTableView->GetFocusedItemData();
+                    if (FocusedApps)
+                    {
+                        if (DownloadApplication(FocusedApps, FALSE))
+                        {
+                            UpdateApplicationsList(-1);
+                        }
+                    }
+                    else
+                    {
+                        // TODO: popup a messagebox telling user to select/check some app first
```

Comment:
Can we disable the button instead?
That is nicer than the user clicking install and getting slapped around with a messagebox

Reply:
yes, we can. But it requires writing more code. I will update the TODO and left it here🤣 (after all, there was even no MessageBox in the old version).